### PR TITLE
WFLY-17964 Upgrade Narayana to 6.0.1.Final

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/WFLY-17964

The component upgrade brings some bug fixes

Tag: https://github.com/jbosstm/narayana/releases/tag/6.0.1.Final  
Diff: https://github.com/jbosstm/narayana/compare/6.0.0.Final...6.0.1.Final 
Release Notes: https://issues.redhat.com/secure/ReleaseNote.jspa?projectId=12310200&version=12402559

### CODE REVIEW

**Comment by bstansberry** [comment]
@xstefank FYI, as the changelog has a bunch of LRA stuff.

**Comment by bstansberry** [comment]
@mmusgrov This has lots of CI failures.

**Comment by mmusgrov** [comment]
> @mmusgrov This has lots of CI failures.

The LRA coordinator isn't starting (NoClassDefFoundError: Could not initialize class io.narayana.lra.coordinator.api.Coordinator).

I followed the README instructions for running the TCK (https://github.com/wildfly/wildfly/blob/main/microprofile/lra/README.md) and the WildFly server fails to start because of an invalid attribute in the subsystem config for microprofile lra in standalone.xml. I corrected that and the server then started OK.

So I am wondering if the standalone.xml for the PR test run has the same problem but I can't figure out what standalone.xml the failed TCK run is using. Can anyone help me figure out how the TCK is configured in the PR?

**Comment by mmusgrov** [comment]
There is another issue reported in the quarkus component upgrade PR ( https://github.com/quarkusio/quarkus/pull/32922#issuecomment-1525596000). For that we need a different image to be pushed to quay.io - will you be able to update https://github.com/jbosstm/lra-coordinator-quarkus to use 6.0.1.Final and then push a new image?

**Comment by mmusgrov** [comment]
Thanks @xstefank I updated my PR adding in the extra module dependency.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
tl;dr; -- CI looks good.

I filed https://issues.redhat.com/browse/WFLY-17968 for the "Linux with security manager" job failure. The "Linux - JDK 17" one is something we've seen elsewhere in the last day and is being investigated; not associated with this PR. The "Windows - JDK 11" one is a long-standing intermittent failure.

**Comment by mmusgrov** [comment]
Great news, thanks Brian

**Review by jamezp — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @mmusgrov, @xstefank and @jamezp
